packages/worker/src/core.ts

```diff
@@ -58,12 +69,40 @@ export class Core {
           : `http://${compiledServerOptions.address}`,
       },
       telemetryOptions,
+      logger: options.logger ?? new DefaultLogger("INFO")
     };
     const native = await promisify(newCore)(compiledOptions);
+    this.initLogPolling(options, native);
+
     return new this(native, compiledOptions);
   }
 
+  private static initLogPolling(options: CoreOptions, native: native.Core) {
+    if (options.logger != null) {
+      interval(3)
+        .pipe(takeUntil(this.shouldPollForLogs.pipe(filter((v) => v === false))))
+        .subscribe((_) => {
+          corePollLogs(native, (_err, logs) => {
+            for (const log of logs) {
+              if (log.level === 'DEBUG') {
+                options.logger?.debug(log.message);
+              } else if (log.level === 'INFO') {
+                options.logger?.info(log.message);
+              } else if (log.level === 'WARN') {
+                options.logger?.warn(log.message);
+              } else if (log.level === 'ERROR') {
+                options.logger?.error(log.message);
+              } else {
+                options.logger?.info(log.message);
+              }
+            }
+          });
+        });
+    }
+  }
+
   protected static _instance?: Promise<Core>;
+  protected static shouldPollForLogs = new BehaviorSubject<boolean>(false);
```

Comment:
Dynamic config?
I probably would attach this as a static attribute, I'd put it on the instance.